registry/handlers/app.go

```diff
@@ -428,6 +428,10 @@ type dispatchFunc func(ctx *Context, r *http.Request) http.Handler
 // handler, using the dispatch factory function.
 func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		for headerName, headerPayload := range app.Config.HTTP.Headers {
```

Comment:
The Add method appends a header. If this used Set instead, duplicate headers would be overwritten instead.

http://golang.org/pkg/net/http/#Header.Add